extensions/sql/contract-definition-store-sql/src/test/java/org/eclipse/dataspaceconnector/sql/contractdefinition/store/SqlContractDefinitionStoreTest.java

```diff
@@ -252,90 +254,117 @@ void findAll_withSpec() {
     }
 
     @Test
-    void findById() {
-        var id = "definitionId";
-        var definition = getContractDefinition(id, "policyId", "contractId");
-        sqlContractDefinitionStore.save(definition);
+    @DisplayName("Find all contract definitions that exactly match a particular access policy ID")
+    void findAll_queryByAccessPolicyId_withEquals() {
 
-        var result = sqlContractDefinitionStore.findById(id);
+        var definitionsExpected = getContractDefinitions(20);
+        sqlContractDefinitionStore.save(definitionsExpected);
 
-        assertThat(result).isNotNull().isEqualTo(definition);
-    }
+        var spec = QuerySpec.Builder.newInstance()
+                .filter("accessPolicyId = policy4")
+                .build();
 
-    @Test
-    void findById_invalidId() {
-        assertThat(sqlContractDefinitionStore.findById("invalid-id")).isNull();
+        var definitionsRetrieved = sqlContractDefinitionStore.findAll(spec).collect(Collectors.toList());
+
+        assertThat(definitionsRetrieved).hasSize(1)
+                .usingRecursiveFieldByFieldElementComparator()
+                .allSatisfy(cd -> assertThat(cd.getId()).isEqualTo("id4"));
     }
 
     @Test
-    void delete() {
-        var definitionExpected = getContractDefinition("test-id1", "policy1", "contract1");
-        sqlContractDefinitionStore.save(definitionExpected);
-        assertThat(sqlContractDefinitionStore.findAll()).hasSize(1);
+    @DisplayName("Find all contract definitions that match a range of access policy IDs")
+    void findAll_queryByAccessPolicyId_withIn() {
 
-        var deleted = sqlContractDefinitionStore.deleteById("test-id1");
-        assertThat(deleted).isNotNull().usingRecursiveComparison().isEqualTo(definitionExpected);
-        assertThat(sqlContractDefinitionStore.findAll()).isEmpty();
-    }
+        var definitionsExpected = getContractDefinitions(20);
+        sqlContractDefinitionStore.save(definitionsExpected);
 
-    @Test
-    void delete_whenNotExist() {
-        var deleted = sqlContractDefinitionStore.deleteById("test-id1");
-        assertThat(deleted).isNull();
+        var spec = QuerySpec.Builder.newInstance()
+                .filter(List.of(new Criterion("accessPolicyId", "in", List.of("policy4", "policy5", "policy6"))))
+                .build();
+
+        var definitionsRetrieved = sqlContractDefinitionStore.findAll(spec).collect(Collectors.toList());
+
+        assertThat(definitionsRetrieved).hasSize(3)
+                .usingRecursiveFieldByFieldElementComparator()
+                .allMatch(cd -> cd.getId().matches("(id)[4-6]"));
     }
 
     @Test
-    void isReferenced_notReferenced() {
-        var definitionsExpected = getContractDefinition("def1", "apol1", "cpol1");
+    @DisplayName("Verify empty result when query contains invalid keys")
```

Comment:
Just a minor thing: this test has the same display name as the following one, even though they test different things.

Reply:
thanks for pointing it out - renamed it.